compiler/src/main/java/com/squareup/anvil/compiler/codegen/CodeGenerator.kt

```diff
@@ -5,6 +5,9 @@ import org.jetbrains.kotlin.psi.KtFile
 import java.io.File
 
 internal interface CodeGenerator {
+
+  val impactsOtherCodeGenerators: Boolean
```

Comment:
This could use some KDoc similar to `flush`. Seems like this encodes whether or not a 2nd pass is required. Would it ever be the case that more than a second pass of the code generators like Dagger would need to be run? If not maybe the name should different. Mostly wondering if there's a more targeted adjective we can use other than `impacts`. No suggestions right now though.

Reply:
Yeah, I'm struggling with the name, too. Code generation happens in rounds. In the next round generators see the generated code of the previous round. In the initial round it's the source code of the module itself. Once no code was generated anymore, then `flush()` is called. 

Some code generated must see the code after calling `flush()`. They get another chance for an extra round, but only if they generate code that no other code generate should process. 

Maybe I should use a separate type instead of a boolean. I think this is cleaner. Suggestions for the name? `PrivateCodeGenerator`, `ConsumingCodeGenerator`?

Reply:
I do like the idea of using the type system for this. `MetaCodeGenerator` could be the type for any `Generator` that wants to operate on previously generated code? Or you use `Tier` or `Pass`: `FirstTierGenerator`, `SecondTierGenerator`.

Maybe this is getting too abstract but while trying to come up with alternatives my mind went to recursion and the distinction between first and higher order functions.

How about `FirstOrderGenerator` for any `Generator` that produces code that is "complete" and does not require further processing by other "higher order generators". `HigherOrderGenerator`: A `Generator` which generates code that could be (must be?) processed / consumed by another `Generator`, first or higher order. This would result in all chains / rounds terminating with `FirstOrderGenerator`s.

The first-order and higher-order naming here changes depending on if the description is being applied to what the `Generator` consumes or produces. In the example above it's aimed at the resulting code but maybe it makes more sense to tie it to what the `Generator` consumes. A la `FirstOrderGenerator` a `Generator` that consumes non-generated code. `HigherOrderGenerator` a `Generator` that consumes code that has been generated and requires further processing.

WDYT?

Reply:
Hm, I personally think it's too abstract 😅  I'll go with `PrivateCodeGenerator` for now. I was considering `SideEffectFreeCodeGenerator`, but that's too long.